### PR TITLE
Finished up the big refactor so vm talks to view with streams now, amoung others

### PR DESCRIPTION
**Pull request by jeffrey-dot-li**
List Of Changes:
feat: update vscout VMs to use streams 
+ feat: add rm functionality
+ feat: add config file functionality (kind of)
+ refactor: make database do multiple actions in one transaction to
increase effeciency and speed
TODO: make config file actually good - make it able to take user input
better, and on start up be able to load differently depending on
different specifications in config.yaml
TODO: create QUERY object to be passed as standard database request
object, similar to RESPONSE
TODO: make all the database commands single transactions (I think only
UPDATE is right now)

### CODE REVIEW

File: bin/vscout.dart [hunk 1]
```diff
@@ -1,54 +1,88 @@
...
 main(List<String> args) async {
-  List<dynamic> commands = List();
+  String absoluteConfigFilePath =
+      ("${dirname(Platform.script.toFilePath()).toString()}${relativeConfigFilePath}");
+  File configFile = File(absoluteConfigFilePath);
+
+  var config = loadYaml(configFile.readAsStringSync());
+
+  print(
+      """
```
Comment (seendsouza) [inline bin/vscout.dart]: Again, dedicated file just for this logo string

File: bin/vscout.dart [hunk 2]
```diff
@@ -1,54 +1,88 @@
...
+                                           /@@@@@@@@@@@@@(                                         
+                                            .@@@@@@@@@@@.                                          
+                                             .@@@@@@@@@,                                           
+                                              .&@@@@@@                                             
+                                                %@@@&                                              
+                                                 %@/                                               
+                                                  .                                                                                                                                                                                          
+  """);
+
+  print('Welcome to vscout cli\n'
```
Comment (seendsouza) [inline bin/vscout.dart]: separate file for vscout will allow this to be all in one print statement

File: lib/src/database/database.dart [hunk 3]
```diff
@@ -94,21 +107,48 @@ class DatabaseHandler {
...
       await this.store.record(entryRecord).update(txn, updateData);
       key = this.store.record(entryRecord).key;
     });
-    results['data'] = key;
-    results['status'] = HttpStatus.ok;
-    return results;
+    return await this.respond(key, 'UPDATE');
+  }
+
+  Future<List<Response>> updateEntries(
```
Comment (seendsouza) [inline lib/src/database/database.dart:124]: will prob change once we go distributed

File: lib/src/views/vscout_view.dart~Stashed changes [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import 'package:flutter/material.dart';
```
Comment (seendsouza) [inline lib/src/views/vscout_view.dart~Stashed changes]: uhhhhhhhhhh sTaShEd ChAnGeS

File: lib/src/utils/parse/parse.dart [hunk 5]
```diff
@@ -14,3 +15,98 @@ parseArgsJson(String str) {
           v is String ? v.trim() : v.toString().trim());
   return result;
 }
+
+List<String> parseArgsList(var string) {
```
Comment (seendsouza) [inline lib/src/utils/parse/parse.dart:19]: so what's wrong with argparse?

**Review by seendsouza — APPROVED** [review]
Rly solid. good work

File: lib/src/utils/parse/parse.dart (hunk 5 above)
Comment (jeffrey-dot-li) [reply lib/src/utils/parse/parse.dart:19]: uhh are you referring to the one from the args package?

Comment (jeffrey-dot-li) [reply lib/src/utils/parse/parse.dart:19]: that one takes a list of strings lol, so I need to build something that parses a string into a list of strings which I feed into the other one

**Pull request merged**